internal/services/iothub/iothub_resource.go

```diff
@@ -376,6 +376,14 @@ func resourceIotHub() *pluginsdk.Resource {
 						},
 
 						"resource_group_name": commonschema.ResourceGroupNameOptional(),
+
+						// NOTE: O+C : required since this property would always be set even if it isn't specified in the tf config, otherwise it would cause a diff and break existing users
+						"subscription_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.IsUUID,
```

Comment:
```suggestion
						"subscription_id": {
							Type:         pluginsdk.TypeString,
							Optional:     true,
							// NOTE: O+C : required since this property would always be set even if it isn't specified in the tf config, otherwise it would cause a diff and break existing users
							Computed:     true,
							ValidateFunc: validation.IsUUID,
```

Reply:
Updated